buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -63,6 +66,30 @@ open class DistributionTest : Test() {
         jvmArgumentProviders.add(libsRepository)
         systemProperty("java9Home", project.findProperty("java9Home") ?: System.getProperty("java9Home"))
     }
+
+    override fun executeTests() {
+        setTimeoutMonitor()
+        super.executeTests()
+    }
+
+    private
+    fun setTimeoutMonitor() {
+        Timer(true).schedule(timerTask {
+            project.javaexec {
+                classpath = project.getConfigurations().getByName("integTestRuntimeClasspath")
+                main = "org.gradle.integtests.fixtures.timeout.JavaProcessStackTracesMonitor"
+            }
+        }, determineTimeoutMillis())
+    }
+
+    private
+    fun determineTimeoutMillis(): Long {
+        return if ("embeded" == System.getProperty("org.gradle.integtest.executer")) {
+            TimeUnit.HOURS.toMillis(1) - TimeUnit.MINUTES.toMillis(1) // in case the build process killed by TC before it finishes
```

Comment:
❌ Our quick feedback builds run at most 15 minutes, so I guess printing the stacktrace after `20` or `30` minutes would be good. See https://e.grdev.net/scans?list.offset=0&list.size=50&list.sortColumn=duration&list.sortOrder=desc&search.startTimeMax=1532436311443&search.startTimeMin=1531831511442&search.tags=QuickFeedback

Reply:
Set to 30 minutes.